_posts/2020-09-29-aws-developer-tools-mule-cicd-solution.md

```diff
@@ -0,0 +1,196 @@
+    anypoint_platform_environment: "Demo"
+    anypoint_platform_workertype: "SMALL"
+    anypoint_platform_workers: "1"
+    anypoint_platform_region: "eu-west-1"
+  parameter-store:
+    anypoint_platform_client_id: "demo-anypoint-client-id"
+    anypoint_platform_client_secret: "demo-anypoint-client-secret"
+
+phases:
+  install:
+    runtime-versions:
+        java: openjdk8
+    commands:
+      - cp ./settings.xml /root/.m2/settings.xml
+      - cp ./settings-security.xml /root/.m2/settings-security.xml
+  build:
+    commands:
+       - cd demo-custom-connector
+       - mvn install -B
+       - cd ../example-service
+       - mvn package deploy -B -DmuleDeploy -Dapp.runtime=$app_runtime -Danypoint.platform.environment=$anypoint_platform_environment -Danypoint.platform.workers=$anypoint_platform_workers -Danypoint.platform.workertype=$anypoint_platform_workertype -Danypoint.platform.region=$anypoint_platform_region -Danypoint.platform.client_id=$anypoint_platform_client_id -Danypoint.platform.client_secret=$anypoint_platform_client_secret
+
+artifacts:
+  files:
+    - demo-custom-connector/target/*.jar
+    - demo-custom-connector/target/*.zip
+
+cache:
+  paths:
+    - '/root/.m2/**/*'
+```
+
+### env-section
+
+This is where local variables are defined for the Build Project and also the variables that are retrieved from Parameter Store. [See Secrets chapter](#secrets). 
+The value for the varible is the key from Parameter Store and CodeBuild replaces the value with the actual value from Parameter Store during execution.
+
+```yml
+env:
+  variables:
+    key: "value"
+  parameter-store:
+    key: "value-name-for-parameter-store"
+```
+
+### phases-section
+
+Phases-section is where the actual build execution is defined. It is split into two sections: install and build.
+
+In the *install*-section the environment for the build is defined. Here I have selected the JDK to use in the build and set custom Maven profile for the build to use.
+The custom settings files are copied from the sources directory (checked out automatically from Git) in the machine running the build to a special location */root/* so that they are taken into use during execution.
+
+In the *build*-section the commands are listed that actually perform the build. For the `mvn package deploy` step the parameters come from the variables defined in the env-section. In the builds stage, I first build and package a custom connector that is used by the MuleSoft application internally.
+
+### artifacts-section
+
+Here is where you can spcify artifacts from the build process to be stored in the S3 bucket. These can then later on be used in deployments to other Amazon Services or other services that can download from S3.
+
+### cache-section
+
+As we all know, Maven tends to download half of the internet to do a build. To speed up builds we should setup a cache. See the [Cache for Maven repository](#cache-for-maven-repository) section to see how to set it up.
```

Comment:
I would avoid using phrases such as "As we all know". Reader might not know it. You could convey the same meaning by saying for example "Notoriously, Maven tends to download half of the internet to do a build."

Reply:
Good point!